### ISSUE TITLE
Searchbox: Clear button has no label

### ISSUE THREAD

**Issue by betrue-final-final** [OfficeDev/office-ui-fabric-react#5025]
Repro:
Launch Narrator
Go to the Search component
Tab into the search box
Right arrow to the clear X button

Result: Narrator says "button"
Expect: Narrator should say "Clear text button"

**Comment by natalieethell** [comment]
Hey @betrue-final-final it looks like for several buttons in other components, we use an optional prop. As an example, in SpinButton, we have the following two optional label texts:

```
  /**
   * Accessibile label text for the increment button for the benefit of the screen reader.
   */
  incrementButtonAriaLabel?: string;

  /**
   * Accessibile label text for the decrement button for the benefit of the screen reader.
   */
  decrementButtonAriaLabel?: string;
```

In Breadcrumb, we have something similar:

```
  /**
   * Optional name to use for aria label on overflow button.
   */
  overflowAriaLabel?: string;
```
Would we like to follow this pattern or just use the string literal 'Clear text button'? Thanks!

**Comment by natalieethell** [comment]
Spoke offline. We'll use the string literal.

**Comment by natalieethell** [comment]
Discussed further with @aditima . It makes more sense to add the aria label string as an optional prop and maybe fallback to the English text. It would be strange to see this aria label show up in en-us if, say, the rest of a webpage is rendering in German.
